node/src/accountant/scanners/scan_schedulers.rs

```diff
@@ -0,0 +1,839 @@
+                    which contradicts the current StartScanError::NothingToProcess"
+                )
+            }
+        } else if err == &StartScanError::NoConsumingWalletFound {
+            SchedulingAfterHaltedScan::Schedule(ScanType::Payables)
+        } else {
+            unreachable!(
+                "{:?} should be impossible with PendingPayableScanner in automatic mode",
+                err
+            )
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::accountant::scanners::scan_schedulers::{
+        HintableScanner, NewPayableScanDynIntervalComputer, NewPayableScanDynIntervalComputerReal,
+        ScanSchedulers, SchedulingAfterHaltedScan,
+    };
+    use crate::accountant::scanners::{MTError, StartScanError};
+    use crate::sub_lib::accountant::ScanIntervals;
+    use itertools::Itertools;
+    use lazy_static::lazy_static;
+    use masq_lib::messages::ScanType;
+    use std::panic::{catch_unwind, AssertUnwindSafe};
+    use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+    #[test]
+    fn scan_schedulers_are_initialized_correctly() {
+        let scan_intervals = ScanIntervals {
+            payable_scan_interval: Duration::from_secs(14),
+            pending_payable_scan_interval: Duration::from_secs(2),
+            receivable_scan_interval: Duration::from_secs(7),
+        };
+        let automatic_scans_enabled = true;
+
+        let schedulers = ScanSchedulers::new(scan_intervals, automatic_scans_enabled);
+
+        assert_eq!(
+            schedulers.payable.nominal_interval,
+            scan_intervals.payable_scan_interval
+        );
+        let payable_scheduler_inner = schedulers.payable.inner.lock().unwrap();
+        assert_eq!(
+            payable_scheduler_inner.last_new_payable_scan_timestamp,
+            UNIX_EPOCH
+        );
+        assert_eq!(
+            schedulers.pending_payable.interval,
+            scan_intervals.pending_payable_scan_interval
+        );
+        assert_eq!(
+            schedulers.receivable.interval,
+            scan_intervals.receivable_scan_interval
+        );
+        assert_eq!(schedulers.automatic_scans_enabled, true)
+    }
+
+    #[test]
+    fn scan_dyn_interval_computer_computes_remaining_time_to_standard_interval_correctly() {
+        let now = SystemTime::now();
+        let inputs = vec![
+            (
+                now.checked_sub(Duration::from_secs(32)).unwrap(),
+                Duration::from_secs(100),
+                Duration::from_secs(68),
+            ),
+            (
+                now.checked_sub(Duration::from_millis(1111)).unwrap(),
+                Duration::from_millis(3333),
+                Duration::from_millis(2222),
+            ),
+            (
+                now.checked_sub(Duration::from_secs(200)).unwrap(),
+                Duration::from_secs(204),
+                Duration::from_secs(4),
+            ),
+        ];
```

Comment:
I think you can refactor a helper a function out of it, which takes the following argument, `past_instant`, `standard_interval`, `expected_result`.